process/factory/indexerFactory.go

```diff
@@ -128,3 +135,21 @@ func checkDataIndexerParams(arguments ArgsIndexerFactory) error {
 
 	return nil
 }
+
+func createBlockCreatorsContainer() (dataindexer.BlockContainerHandler, error) {
```

Comment:
this could have been part of the core, exposed via a function `CreateDefaultBlockCreatorsContainer()`

Reply:
We had something similar in the `mx-chain-core-go` repository but after some discussions, the `createBlockCreatorsContainer` should stay in clients in case a client will have a different header type.